File: Clients/BlogBlazor/Shared/Blazor.Contracts/V1/Responses/PostsResponses/PostWithPagedCommentsResponse.cs
namespace Blazor.Contracts.V1.Responses.PostsResponses;

using System.Collections.Generic;
using CommentsResponses;
using TagsResponses;

/// <summary>
/// Post with paged comments response.
/// </summary>
public class PostWithPagedCommentsResponse
{
    /// <summary>
    /// Gets or sets post.
    /// </summary>
    public PostViewResponse Post { get; set; }

    /// <summary>
    /// Gets or sets the comments.
    /// </summary>
    /// <value>
    /// The comments.
    /// </value>
    public PagedCommentsResponse Comments { get; set; }

    // public Comment Comment { get; set; }
    // public int CommentsCount { get; set; }
    // public Profile Profile { get; set; }

    /// <summary>
    /// Gets or sets the tags.
    /// </summary>
    /// <value>
    /// The tags.
    /// </value>
    public IList<TagResponse> Tags { get; set; }
}

File: Shared/Blog.FSharp.Contracts/V1/Responses/PostsResponses/PostWithPagedCommentsResponse.fs
namespace Blog.FSharp.Contracts.V1.Responses.PostsResponses

open System.Collections.Generic
open Blog.FSharp.Contracts.V1.Responses.CommentsResponses
open Blog.FSharp.Contracts.V1.Responses.TagsResponses

/// <summary>
/// Post with paged comments response.
/// </summary>
type PostWithPagedCommentsResponse() =
    /// <summary>
    /// Gets or sets post.
    /// </summary>
    member val Post: PostViewResponse = PostViewResponse() with get, set

    /// <summary>
    /// Gets or sets the comments.
    /// </summary>
    member val Comments: PagedCommentsResponse = PagedCommentsResponse() with get, set

    /// <summary>
    /// Gets or sets the tags.
    /// </summary>
    member val Tags: IList<TagResponse> = List<TagResponse>() :> IList<TagResponse> with get, set

